src/realm/util/system_process.cpp

```diff
@@ -364,12 +364,24 @@ auto sys_proc::copy_local_environment() -> Environment
         throw std::runtime_error("Not yet implemented");
 
 #else // defined _WIN32 && !REALM_UWP
+    auto wstring_to_utf8 = [](wchar_t* w_str) {
+        // First get the number of chars needed for output buffer
+        int chars_num = WideCharToMultiByte(CP_UTF8, 0, w_str, -1, nullptr, 0, nullptr, nullptr);
+        char* str = new char[chars_num];
```

Comment:
Why not `auto str = std::unique_ptr<char[]>(new char[chars_num]);`?  Or even better, `auto result = std::string(chars_num);` so that the conversion is done directly into the destination buffer? If you go that route, you will either need to adjust the sizes so that it doesn't have a double null termination, or just do `REALM_ASSERT(result.back() == '\0'); result.resize(result.size()-1);` to remove it after the fact.

Reply:
I never got back to this. Not sure if it even works. Anyway - it does not seem to be used. Will probably just delete the whole module. Asking @simonask,